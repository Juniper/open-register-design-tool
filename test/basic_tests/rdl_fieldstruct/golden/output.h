//   Ordt 230719.01 autogenerated file 
//   Input: ./rdl_fieldstruct/test.rdl
//   Parms: ./rdl_fieldstruct/test.parms
//   Date: Thu Jul 20 13:46:42 EDT 2023
//

#ifndef __FOO_REGISTER_MAP__
#define __FOO_REGISTER_MAP__

#include <bits.h>

/* FOO_REGISTERS memory map */
enum FOO_REGS {
  AREG_0 = 0x0,
  AREG_1 = 0x4,
  BLABLA = 0x8
};

/* AREG_0 register fields */
#define AREG_0_HIER_FS_FS1_FLD1 GENMASK(3, 0)
#define AREG_0_HIER_FS_FS1_FLD2 BIT(4)
#define AREG_0_HIER_FS_FLD1 BIT(5)
#define AREG_0_HIER_FS_FS2_FLD1 GENMASK(9, 6)
#define AREG_0_HIER_FS_FS2_FLD2 BIT(14)
#define AREG_0_FS3_FLD1 GENMASK(27, 24)
#define AREG_0_FS3_FLD2 BIT(28)

/* AREG_1 register fields */
#define AREG_1_HIER_FS_FS1_FLD1 GENMASK(3, 0)
#define AREG_1_HIER_FS_FS1_FLD2 BIT(4)
#define AREG_1_HIER_FS_FLD1 BIT(5)
#define AREG_1_HIER_FS_FS2_FLD1 GENMASK(9, 6)
#define AREG_1_HIER_FS_FS2_FLD2 BIT(14)
#define AREG_1_FS3_FLD1 GENMASK(27, 24)
#define AREG_1_FS3_FLD2 BIT(28)

/* BLABLA register fields */
#define BLABLA_FS1_0_FLD1 GENMASK(3, 0)
#define BLABLA_FS1_0_FLD2 BIT(4)
#define BLABLA_FS1_1_FLD1 GENMASK(8, 5)
#define BLABLA_FS1_1_FLD2 BIT(9)
#define BLABLA_FS1_2_FLD1 GENMASK(13, 10)
#define BLABLA_FS1_2_FLD2 BIT(14)
#define BLABLA_FS3_0_FLD1 GENMASK(18, 15)
#define BLABLA_FS3_0_FLD2 BIT(19)
#define BLABLA_FS3_1_FLD1 GENMASK(26, 23)
#define BLABLA_FS3_1_FLD2 BIT(27)

#endif
