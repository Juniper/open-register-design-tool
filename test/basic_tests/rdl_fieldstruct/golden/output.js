/*   Ordt 190725.01 autogenerated file 
 *   Input: ./rdl_fieldstruct/test.rdl
 *   Parms: ./rdl_fieldstruct/test.parms
 *   Date: Sun Nov 17 17:00:57 EST 2019
 */

register_set foo "Registers for foo" {

  address = 0x0;
  register_width = 32;
  register areg "areg register" {
    address = 0x0;
    repeat = 2;
    nop[3];
    integer fs3_fld2[1] "fs3_fld2 field" param {
      access_mode = READ_WRITE;
      reset = unknown;
    };
    integer fs3_fld1[4] "fs3_fld1 field" param {
      access_mode = READ_WRITE;
      reset = 0x0;
    };
    nop[9];
    integer hier_fs_fs2_fld2[1] "hier_fs_fs2_fld2 field" param {
      access_mode = READ_WRITE;
      reset = unknown;
    };
    nop[4];
    integer hier_fs_fs2_fld1[4] "hier_fs_fs2_fld1 field" param {
      access_mode = READ_WRITE;
      reset = 0x0;
    };
    integer hier_fs_fld1[1] "hier_fs_fld1 field" param {
      access_mode = READ_WRITE;
      reset = unknown;
    };
    integer hier_fs_fs1_fld2[1] "hier_fs_fs1_fld2 field" param {
      access_mode = READ_WRITE;
      reset = unknown;
    };
    integer hier_fs_fs1_fld1[4] "hier_fs_fs1_fld1 field" param {
      access_mode = READ_WRITE;
      reset = 0x0;
    };
  };
  
  register blabla "blabla register" {
    address = 0x8;
    nop[4];
    integer fs3_1_fld2[1] "fs3_1_fld2 field" param {
      access_mode = READ_WRITE;
      reset = unknown;
    };
    integer fs3_1_fld1[4] "fs3_1_fld1 field" param {
      access_mode = READ_WRITE;
      reset = 0x0;
    };
    nop[3];
    integer fs3_0_fld2[1] "fs3_0_fld2 field" param {
      access_mode = READ_WRITE;
      reset = unknown;
    };
    integer fs3_0_fld1[4] "fs3_0_fld1 field" param {
      access_mode = READ_WRITE;
      reset = 0x0;
    };
    integer fs1_2_fld2[1] "fs1_2_fld2 field" param {
      access_mode = READ_WRITE;
      reset = unknown;
    };
    integer fs1_2_fld1[4] "fs1_2_fld1 field" param {
      access_mode = READ_WRITE;
      reset = 0x0;
    };
    integer fs1_1_fld2[1] "fs1_1_fld2 field" param {
      access_mode = READ_WRITE;
      reset = unknown;
    };
    integer fs1_1_fld1[4] "fs1_1_fld1 field" param {
      access_mode = READ_WRITE;
      reset = 0x0;
    };
    integer fs1_0_fld2[1] "fs1_0_fld2 field" param {
      access_mode = READ_WRITE;
      reset = unknown;
    };
    integer fs1_0_fld1[4] "fs1_0_fld1 field" param {
      access_mode = READ_WRITE;
      reset = 0x0;
    };
  };
  
  register_set_size = 0xc;
};

