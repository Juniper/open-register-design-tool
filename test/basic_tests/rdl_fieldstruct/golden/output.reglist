//   Ordt 190725.01 autogenerated file 
//   Input: ./rdl_fieldstruct/test.rdl
//   Parms: ./rdl_fieldstruct/test.parms
//   Date: Sun Nov 17 17:02:49 EST 2019
//

0x0              size: 32   fields: 7    bits: 16        areg_0
  hier_fs_fs1_fld1 [3:0]     mode: RW   reset: 4'd0                  ( areg[0].hier_fs_fs1_fld1 = 4'd0 )
  hier_fs_fs1_fld2 [4]       mode: RW   reset:                       ( areg[0].hier_fs_fs1_fld2 )
  hier_fs_fld1 [5]           mode: RW   reset:                       ( areg[0].hier_fs_fld1 )
  hier_fs_fs2_fld1 [9:6]     mode: RW   reset: 4'd0                  ( areg[0].hier_fs_fs2_fld1 = 4'd0 )
  hier_fs_fs2_fld2 [14]      mode: RW   reset:                       ( areg[0].hier_fs_fs2_fld2 )
  fs3_fld1 [27:24]           mode: RW   reset: 4'd0                  ( areg[0].fs3_fld1 = 4'd0 )
  fs3_fld2 [28]              mode: RW   reset:                       ( areg[0].fs3_fld2 )
0x4              size: 32   fields: 7    bits: 16        areg_1
  hier_fs_fs1_fld1 [3:0]     mode: RW   reset: 4'd0                  ( areg[1].hier_fs_fs1_fld1 = 4'd0 )
  hier_fs_fs1_fld2 [4]       mode: RW   reset:                       ( areg[1].hier_fs_fs1_fld2 )
  hier_fs_fld1 [5]           mode: RW   reset:                       ( areg[1].hier_fs_fld1 )
  hier_fs_fs2_fld1 [9:6]     mode: RW   reset: 4'd0                  ( areg[1].hier_fs_fs2_fld1 = 4'd0 )
  hier_fs_fs2_fld2 [14]      mode: RW   reset:                       ( areg[1].hier_fs_fs2_fld2 )
  fs3_fld1 [27:24]           mode: RW   reset: 4'd0                  ( areg[1].fs3_fld1 = 4'd0 )
  fs3_fld2 [28]              mode: RW   reset:                       ( areg[1].fs3_fld2 )
0x8              size: 32   fields: 10   bits: 25        blabla
  fs1_0_fld1 [3:0]           mode: RW   reset: 4'd0                  ( blabla.fs1_0_fld1 = 4'd0 )
  fs1_0_fld2 [4]             mode: RW   reset:                       ( blabla.fs1_0_fld2 )
  fs1_1_fld1 [8:5]           mode: RW   reset: 4'd0                  ( blabla.fs1_1_fld1 = 4'd0 )
  fs1_1_fld2 [9]             mode: RW   reset:                       ( blabla.fs1_1_fld2 )
  fs1_2_fld1 [13:10]         mode: RW   reset: 4'd0                  ( blabla.fs1_2_fld1 = 4'd0 )
  fs1_2_fld2 [14]            mode: RW   reset:                       ( blabla.fs1_2_fld2 )
  fs3_0_fld1 [18:15]         mode: RW   reset: 4'd0                  ( blabla.fs3_0_fld1 = 4'd0 )
  fs3_0_fld2 [19]            mode: RW   reset:                       ( blabla.fs3_0_fld2 )
  fs3_1_fld1 [26:23]         mode: RW   reset: 4'd0                  ( blabla.fs3_1_fld1 = 4'd0 )
  fs3_1_fld2 [27]            mode: RW   reset:                       ( blabla.fs3_1_fld2 )
