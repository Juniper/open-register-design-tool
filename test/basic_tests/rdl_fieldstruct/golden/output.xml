<?xml version="1.0" encoding="UTF-8"?>
<!--   Ordt 190725.01 autogenerated file 
       Input: ./rdl_fieldstruct/test.rdl
       Parms: ./rdl_fieldstruct/test.parms
       Date: Sun Nov 17 17:08:17 EST 2019
 -->

<map version="190725.01">
  <id>foo</id>
  <baseaddr>0</baseaddr>
  <shorttext>foo registers</shorttext>
  <reg>
    <id>areg</id>
    <shorttext>areg register</shorttext>
    <baseaddr>0x0</baseaddr>
    <width>32</width>
    <reps>2</reps>
    <stride>0x4</stride>
    <field>
      <id>hier_fs_fs1_fld1</id>
      <shorttext>hier_fs_fs1_fld1 field</shorttext>
      <access>RW</access>
      <hwinfo>
        <hwaccess>rw</hwaccess>
        <we></we>
      </hwinfo>
      <reset>0</reset>
      <lowidx>0</lowidx>
      <width>4</width>
      <hwmod></hwmod>
    </field>
    <field>
      <id>hier_fs_fs1_fld2</id>
      <shorttext>hier_fs_fs1_fld2 field</shorttext>
      <access>RW</access>
      <hwinfo>
        <hwaccess>r</hwaccess>
      </hwinfo>
      <lowidx>4</lowidx>
      <width>1</width>
    </field>
    <field>
      <id>hier_fs_fld1</id>
      <shorttext>hier_fs_fld1 field</shorttext>
      <access>RW</access>
      <hwinfo>
        <hwaccess>r</hwaccess>
      </hwinfo>
      <lowidx>5</lowidx>
      <width>1</width>
    </field>
    <field>
      <id>hier_fs_fs2_fld1</id>
      <shorttext>hier_fs_fs2_fld1 field</shorttext>
      <access>RW</access>
      <hwinfo>
        <hwaccess>rw</hwaccess>
        <we></we>
      </hwinfo>
      <reset>0</reset>
      <lowidx>6</lowidx>
      <width>4</width>
      <hwmod></hwmod>
    </field>
    <field>
      <id>hier_fs_fs2_fld2</id>
      <shorttext>hier_fs_fs2_fld2 field</shorttext>
      <access>RW</access>
      <hwinfo>
        <hwaccess>r</hwaccess>
      </hwinfo>
      <lowidx>14</lowidx>
      <width>1</width>
    </field>
    <field>
      <id>fs3_fld1</id>
      <shorttext>fs3_fld1 field</shorttext>
      <access>RW</access>
      <hwinfo>
        <hwaccess>rw</hwaccess>
        <we></we>
      </hwinfo>
      <reset>0</reset>
      <lowidx>24</lowidx>
      <width>4</width>
      <hwmod></hwmod>
    </field>
    <field>
      <id>fs3_fld2</id>
      <shorttext>fs3_fld2 field</shorttext>
      <access>RW</access>
      <hwinfo>
        <hwaccess>r</hwaccess>
      </hwinfo>
      <lowidx>28</lowidx>
      <width>1</width>
    </field>
    <access>RW</access>
  </reg>
  <reg>
    <id>blabla</id>
    <shorttext>blabla register</shorttext>
    <baseaddr>0x8</baseaddr>
    <width>32</width>
    <field>
      <id>fs1_0_fld1</id>
      <shorttext>fs1_0_fld1 field</shorttext>
      <access>RW</access>
      <hwinfo>
        <hwaccess>rw</hwaccess>
        <we></we>
      </hwinfo>
      <reset>0</reset>
      <lowidx>0</lowidx>
      <width>4</width>
      <hwmod></hwmod>
    </field>
    <field>
      <id>fs1_0_fld2</id>
      <shorttext>fs1_0_fld2 field</shorttext>
      <access>RW</access>
      <hwinfo>
        <hwaccess>r</hwaccess>
      </hwinfo>
      <lowidx>4</lowidx>
      <width>1</width>
    </field>
    <field>
      <id>fs1_1_fld1</id>
      <shorttext>fs1_1_fld1 field</shorttext>
      <access>RW</access>
      <hwinfo>
        <hwaccess>rw</hwaccess>
        <we></we>
      </hwinfo>
      <reset>0</reset>
      <lowidx>5</lowidx>
      <width>4</width>
      <hwmod></hwmod>
    </field>
    <field>
      <id>fs1_1_fld2</id>
      <shorttext>fs1_1_fld2 field</shorttext>
      <access>RW</access>
      <hwinfo>
        <hwaccess>r</hwaccess>
      </hwinfo>
      <lowidx>9</lowidx>
      <width>1</width>
    </field>
    <field>
      <id>fs1_2_fld1</id>
      <shorttext>fs1_2_fld1 field</shorttext>
      <access>RW</access>
      <hwinfo>
        <hwaccess>rw</hwaccess>
        <we></we>
      </hwinfo>
      <reset>0</reset>
      <lowidx>10</lowidx>
      <width>4</width>
      <hwmod></hwmod>
    </field>
    <field>
      <id>fs1_2_fld2</id>
      <shorttext>fs1_2_fld2 field</shorttext>
      <access>RW</access>
      <hwinfo>
        <hwaccess>r</hwaccess>
      </hwinfo>
      <lowidx>14</lowidx>
      <width>1</width>
    </field>
    <field>
      <id>fs3_0_fld1</id>
      <shorttext>fs3_0_fld1 field</shorttext>
      <access>RW</access>
      <hwinfo>
        <hwaccess>rw</hwaccess>
        <we></we>
      </hwinfo>
      <reset>0</reset>
      <lowidx>15</lowidx>
      <width>4</width>
      <hwmod></hwmod>
    </field>
    <field>
      <id>fs3_0_fld2</id>
      <shorttext>fs3_0_fld2 field</shorttext>
      <access>RW</access>
      <hwinfo>
        <hwaccess>r</hwaccess>
      </hwinfo>
      <lowidx>19</lowidx>
      <width>1</width>
    </field>
    <field>
      <id>fs3_1_fld1</id>
      <shorttext>fs3_1_fld1 field</shorttext>
      <access>RW</access>
      <hwinfo>
        <hwaccess>rw</hwaccess>
        <we></we>
      </hwinfo>
      <reset>0</reset>
      <lowidx>23</lowidx>
      <width>4</width>
      <hwmod></hwmod>
    </field>
    <field>
      <id>fs3_1_fld2</id>
      <shorttext>fs3_1_fld2 field</shorttext>
      <access>RW</access>
      <hwinfo>
        <hwaccess>r</hwaccess>
      </hwinfo>
      <lowidx>27</lowidx>
      <width>1</width>
    </field>
    <access>RW</access>
  </reg>
</map>
